### PR TITLE
'openssl list': add option -objects to list built in objects

### PR DESCRIPTION
**Pull request by levitte**
Related to #6696

### CODE REVIEW

**Review by richsalz — COMMENTED** [review]
list.pod needs to be updated.

**Comment by levitte** [comment]
Done.  There should be an entry i CHANGES, shouldn't there?  Also, a question we need to ask ourselves is if this is fit for 1.1.1 or after.

**Comment by richsalz** [comment]
no opinion on 1.1.1 or later.  hard to see it as a bug-fix.

File: apps/openssl.c [hunk 1]
```diff
@@ -316,6 +316,37 @@ static void list_missing_help(void)
...
 
+static void list_objects(void)
+{
+    int max_nid = OBJ_new_nid(0);
+    int i;
+
+    /* Skip 0, since that's NID_undef */
+    for (i = 1; i < max_nid; i++) {
+        const ASN1_OBJECT *obj = OBJ_nid2obj(i);
+        char oid_buf[128];       /* I dare you to find such a long OID */
```
Comment (paulidale) [inline apps/openssl.c]: A challenge is a bad thing to pose a bored programmer.

**Review by paulidale — APPROVED** [review]
I'm not fussed either way about a CHANGES entry.

I agree that this is a new feature and thus should be post 1.1.1.  However, it also looks pretty harmless and self contained.

File: apps/openssl.c (hunk 1 above)
Comment (levitte) [reply apps/openssl.c]: I think you'll find that `OBJ_create` call quite a challenge in itself ;-)

Comment (levitte) [reply apps/openssl.c]: But sure, someone might do `OBJ_create("2.1.20.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999", "myShortName", "My Longer Name!");`

Comment (levitte) [reply apps/openssl.c]: The problem here is that `OBJ_obj2txt` will silently stop filling the buffer when its size has been reached, so while a limited buffer is without harm, the OID output may be truncated, possibly invisibly.  Fixing that is a matter for a different PR, methinks.

Comment (levitte) [reply apps/openssl.c]: Oh!  No, I was mistaken...  `OBJ_obj2txt` returns the size the buffer *should* have.  Ok, will change the buffer to be heap-reallocatable.

**Comment by mattcaswell** [comment]
Can this be merged now?

**Comment by levitte** [comment]
Thanks for the reminder.  I decided to add a small entry in CHANGES, so this needs reconfirmation.

**Comment by levitte** [comment]
This still needs reconfirming

**Review by t-j-h — APPROVED** [review]

**Comment by levitte** [comment]
Merged.

8ddbff9c0811a0f11855eda871b9d3bff8fb325e 'openssl list': add option -objects to list built in objects

**Pull request closed without merging**